src/value/mod.rs

```diff
@@ -167,6 +185,50 @@ pub(crate) fn mg_value_string(mg_value: *const bindings::mg_value) -> String {
     mg_string_to_string(c_str)
 }
 
+pub(crate) fn mg_value_naive_date(mg_value: *const bindings::mg_value) -> Option<NaiveDate> {
+    let c_date = unsafe { bindings::mg_value_date(mg_value) };
+    let c_delta_days = unsafe { bindings::mg_date_days(c_date) };
+    let epoch_date = NaiveDate::from_ymd(1970, 1, 1);
+    let delta_days = Duration::days(c_delta_days);
+    epoch_date.checked_add_signed(delta_days)
+}
+
+pub(crate) fn mg_value_naive_local_time(mg_value: *const bindings::mg_value) -> Option<NaiveTime> {
+    let nsec_in_sec = 1_000_000_000;
+    let u32_max = u32::MAX as i64;
+    let c_local_time = unsafe { bindings::mg_value_local_time(mg_value) };
+    let c_nanoseconds = unsafe { bindings::mg_local_time_nanoseconds(c_local_time) };
+    let seconds = c_nanoseconds / nsec_in_sec;
+    let nanoseconds = c_nanoseconds % nsec_in_sec;
+    if seconds > u32_max {
+        return None;
+    }
+    if nanoseconds > u32_max {
+        return None;
+    }
```

Comment:
Does [TryForm](https://doc.rust-lang.org/std/convert/trait.TryFrom.html) work here? :thinking:

Reply:
Probably!

Reply:
This
```
pub(crate) fn mg_value_naive_local_time(mg_value: *const bindings::mg_value) -> Option<NaiveTime> {
    let nsec_in_sec = 1_000_000_000;
    let c_local_time = unsafe { bindings::mg_value_local_time(mg_value) };
    let c_nanoseconds = unsafe { bindings::mg_local_time_nanoseconds(c_local_time) };
    let seconds = match u32::try_from(c_nanoseconds / nsec_in_sec) {
        Ok(x) => x,
        Err(_) => { return None; }
    };
    let nanoseconds = match u32::try_from(c_nanoseconds % nsec_in_sec) {
        Ok(x) => x,
        Err(_) => { return None; }
    };
    NaiveTime::from_num_seconds_from_midnight_opt(seconds, nanoseconds)
}
```
works, but it's not nice, forwarding Result with `?` operator is also not trivial (I don't know how to figure out the `Error` type) + even if I figure out, the last line is returning an optional value (I have to deal with that)...

Reply:
I'm ok with the old solution or the one I've posted here :smile: It's a bit more code, but better code :smile: @antonio2368 what should stay? :smile:

Reply:
I'm not sure I follow but you can't do:
```
let seconds = u32::try_from(c_nanoseconds % nsec_in_sec)?;
```
?
It's nothing critical you can leave everything as is. First time checking out more in depth Rust so everything is just interesting to me.

Reply:
Also, even if it's more code, I still prefer the above example because you don't have to write logic yourself.

Reply:
Only `?` is not applicable because the return value is `Option<>`, only `Result<>` is supported... The problem with Result is that I don't know how to specify error type because each TryFrom has a different error, I'm sure that's somehow possible but :timer_clock: + the last line is then outlier because they decided to handle error by returning optional :joy:

Reply:
I've managed to make `try_from` work :smile: @antaljanosbenjamin take a look now :smile: Maybe for @antaljanosbenjamin are these `unwraps` fine or we propagate the exact error somehow (I'm more keen into polishing that a bit)?

Reply:
I have to join the Rust club :D